include/fifo.h

```diff
@@ -49,10 +49,14 @@
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
 
+/**
+ * @struct fifo_element
+ * @brief Structure holding the fifo element parameters.
+ */
 struct fifo_element {
-	struct fifo_element *next;
-	char *data;
-	uint32_t len;
+	struct fifo_element *next; ///< next FIFO element
```

Comment:
Why did you decide to place the documentation block after the member (shouldn't we do it before, the same as for the function's parameters for example)?

Reply:
I forgot to mention in the PR description that this is a proposal, not a decision. This approach populates the already available Doxygen fields (Member Data Documentation/Enumeration Type Documentation) with the inserted comments VS original approach which keeps everything in the brief description of the structure/enum. So it depends on how "sophisticated" we want the documentation to be.

Reply:
Update: I added an alternative that comments the members above the implementation. Please see #390. After a decision is made, this PR can be closed.